### PR TITLE
Automatically port Reflection.PortableExecutable.DllCharacteristics

### PR DESCRIPTION
**Pull request by carlossanlop**

### CODE REVIEW

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
       </ReturnValue>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Can we add a type summary, something like "Describes the characteristics of a dynamic link library."?

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml [hunk 2]
```diff
@@ -348,7 +348,7 @@
...
       <MemberValue>8192</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Driver uses WDM model.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]:
```suggestion
        <summary>The driver uses the WDM model.</summary>
```

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml [hunk 3]
```diff
@@ -248,7 +248,7 @@
       </ReturnValue>
       <MemberValue>2</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Reserved.</summary>
       </Docs>
     </Member>
     <Member MemberName="TerminalServerAware">
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Do we have a description for TerminalServerAware?

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml [hunk 4]
```diff
@@ -74,7 +74,7 @@
       </ReturnValue>
       <MemberValue>64</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>DLL can move.</summary>
```
Comment (rpetrusha) [inline xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]:
```suggestion
        <summary>The DLL can move.</summary>
```

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Not in the source comments: [src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEFileFlags.cs#L113](https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEFileFlags.cs#L113)

This enum value was introduced in the very first commit of .NET Core: https://github.com/dotnet/corefx/commit/f11305d31a07ef9a2513332edc665b257cc9c5e5

So the API is as old as Framework. But judging by the value, and a quick search I did for these terms, this is my proposed description, @rpetrusha:

```The image is Terminal Server aware.```

@GrabYourPitchforks @steveharter as the area owners, is this description good enough for you?

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: The type summary sounds good to me, I will apply it.
@GrabYourPitchforks @steveharter as the area owners, do you agree with the type summary?

Comment (carlossanlop) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Actually, @tmat please take a look at this PR as well.

Comment (tmat) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: lgtm

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml [hunk 5]
```diff
@@ -74,7 +74,7 @@
       </ReturnValue>
       <MemberValue>64</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The DLL can move.</summary>
```
Comment (sdmaclea) [inline xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: This doesn't seem to make sense.

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml (hunk 3 above)
Comment (sdmaclea) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Seems reasonable to me, but I am not an expert.

File: xml/System.Reflection.PortableExecutable/DllCharacteristics.xml (hunk 5 above)
Comment (sdmaclea) [reply xml/System.Reflection.PortableExecutable/DllCharacteristics.xml]: Maybe it should be `The DLL can be relocated.`

**Comment by rpetrusha** [comment]
I think that this is ready to merge, @carlossanlop. I've changed "The DLL can move" to "The DLL can be relocated." I've approved, so let me know if it's ready to merge.

**Comment by carlossanlop** [comment]
Your last PR is ready to merge, @rpetrusha :)

**Comment by rpetrusha** [comment]
Thanks, @carlossanlop. I'll merge now.

**Pull request merged**